rotkehlchen/inquirer.py

```diff
@@ -890,12 +888,43 @@ def find_yearn_price(
         ethereum = self.get_ethereum_manager()
         globaldb = GlobalDBHandler()
         with globaldb.conn.read_ctx() as cursor:
-            maybe_underlying_token = globaldb.fetch_underlying_tokens(cursor, ethaddress_to_identifier(token.evm_address))  # noqa: E501
-        if maybe_underlying_token is None or len(maybe_underlying_token) != 1:
-            log.error(f'Yearn vault token {token} without an underlying asset')
-            return None
+            maybe_underlying_tokens = globaldb.fetch_underlying_tokens(cursor, ethaddress_to_identifier(token.evm_address))  # noqa: E501
+
+        contract = EvmContract(
+            address=token.evm_address,
+            abi=ethereum.node_inquirer.contracts.abi('YEARN_VAULT_V2'),
+            deployed_block=0,
+        )
+        if maybe_underlying_tokens is None or len(maybe_underlying_tokens) != 1:
+            # underlying token not recorded in the DB. Ask the chain
+            try:
+                remote_underlying_token = contract.call(ethereum.node_inquirer, 'token')
+            except (RemoteError, BlockchainQueryError) as e:
+                log.error(f'Failed to query underlying token method in Yearn v2 Vault. {str(e)}')
+                return None
+
+            try:
+                underlying_token_address = deserialize_evm_address(remote_underlying_token)
```

Comment:
Shouldn't this also do a get_or_create evm token? It might be the case that the underlying token is not in the globaldb for some reason